src/pynxtools/nomad/parser.py

```diff
@@ -472,6 +484,23 @@ def parse(
                 break
         archive.metadata.entry_type = app_def
 
+        # group parsed content from all appdefs of a scientific domain to metainfo.domain
+        scientific_domain: list[str] = []
+        for domain_key, domain_val in NXDEF_TO_METAINFO_DOMAIN.items():
+            if isinstance(domain_val, list):
+                for prefix in domain_val:
+                    if app_def.startswith(prefix):
+                        scientific_domain.append(domain_key)
+        if len(scientific_domain) == 1:
+            # NeXusParser(Parser), domain is member of parser, therein initialized to 'dft'
+            archive.metainfo.domain = scientific_domain[0]
+        else:
+            archive.metainfo.domain = "exp"
+            if len(scientific_domain) > 1:
+                self._logger.warn(
+                    f"cannot resolve domain for multi-domain {scientific_domain}"
+                )
+
```

Comment:
I suggest to switch the key and value in the lookup table and directly check it. Also I think it makes sense to explicitly spell out all sub-appdefs as we agreed against the prefixing naming convention (with `NXmpes_arpes` being the exception because of the existence of `NXarpes`).

```suggestion
        if appdef in NXDEF_TO_METAINFO_DOMAIN:
           archive.metainfo.domain = NXDEF_TO_METAINFO_DOMAIN[appdef]
        else:
          archive.metainfo.domain = "exp"
          ... do logging (but I think it's not really necessary as we fall back to a reasonable default)
```

Reply:
Agree to your proposal of simplifying the lookup, implemented this.

I have a different opinion about the metainfo concepts "entry_type" and "domain".
Currently, the nexus parser has populated entry_type already with the name of the appdef which is useful as it quickly allows to filter via ES for specific measurements following specific appdefs. "Domain" indeed if the mapping table is always each key has only one value then there is not much of an advantage. However, currently we have e.g. the EELSDB whose domain is 'ems' (likely a shorthand for electron microscopy spectroscopy) it is a poor enum value but it has a benefit namely to group entries related to electron microscopy which is not only NXem, domain can be one such tag-like mechanism whereby different entries within the same scientific domain can be filtered for irrespective whether they have been generated using the NeXus parser or e.g. EELS or something maybe from custom base sections within customized parsing. I agree that this is just one tag and thus limited and like with many discussions about suitable tags there will be then people asking for more fine granularizable tags, well its a tradeoff, indeed in what NOMAD was so far (heavily biased towards dft) why to have at all a domain? Now with reaching out to experiments and many different methods I think this is really useful, not only because in fairmat we have not yet been able to agree that even for em there is at least one mechanism where all em related entries can be found by. Domain is one such in my opinion, I agree maybe not an ideal one for everyones particularity interests and expectation how they are using nomad